src/client.rs

```diff
@@ -100,140 +109,417 @@ pub enum ApiError {
     },
 }
 
-type ClientResult<T> = Result<T, ClientError>;
-
 /// Spotify API object
-#[derive(Debug, Clone)]
+#[derive(Builder, Debug, Clone)]
 pub struct Spotify {
-    client: Client,
+    /// reqwest needs an instance of its client to perform requests.
+    #[cfg(feature = "client-reqwest")]
+    #[builder(setter(skip))]
+    pub(in crate) client: reqwest::Client,
+
+    /// The access token information required for requests to the Spotify API.
+    #[builder(setter(strip_option), default)]
+    pub token: Option<Token>,
+
+    /// The credentials needed for obtaining a new access token, for requests
+    /// without OAuth authentication.
+    #[builder(setter(strip_option), default)]
+    pub credentials: Option<Credentials>,
+
+    /// The OAuth information required for obtaining a new access token, for
+    /// requests with OAuth authentication. `credentials` also needs to be
+    /// set up.
+    #[builder(setter(strip_option), default)]
+    pub oauth: Option<OAuth>,
+
+    /// The Spotify API prefix, `https://api.spotify.com/v1/` by default.
+    #[builder(
+        setter(into),
+        default = "String::from(\"https://api.spotify.com/v1/\")"
```

Comment:
how about using `r##r` operator instead of escaped string?

Reply:
Good idea, I'll fix that.